### PR TITLE
Fix ubuntu build

### PR DESCRIPTION
**Pull request by devhawk**
changes to enable xlang tools, PAL and to compile on Ubuntu 18.04 with clang 6. Tested via Windwos Susbsytem for Linux.

Note, convert_string functions in PAL remain unimplemented on non-windows platforms. As such, 4 of the PAL tests fail on Ubuntu. #47 opened to track the issue, assigned to @DefaultRyan.

### CODE REVIEW

File: src/platform/common_string_convert.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    uint32_t convert_string(
+        char16_t const* input_str,
+        uint32_t input_length,
+        xlang_char8* output_buffer,
+        uint32_t buffer_length)
+    {
+        static_assert(sizeof(xlang_char8) == sizeof(char));
+ 
+        // TODO: implement convert_string for non-windows platforms (#47)
```
Comment (Scottj1s) [inline src/platform/common_string_convert.cpp]: Remove comment?

File: src/platform/common_string_convert.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    
+    uint32_t convert_string(
+        xlang_char8 const* input_str,
+        uint32_t input_length,
+        char16_t* output_buffer,
+        uint32_t buffer_length)
+    {
+        static_assert(sizeof(xlang_char8) == sizeof(char));
+
+        // TODO: implement convert_string for non-windows platforms (#47)
```
Comment (Scottj1s) [inline src/platform/common_string_convert.cpp]: ditto

File: src/platform/common_string_convert.cpp (hunk 1 above)
Comment (devhawk) [reply src/platform/common_string_convert.cpp]: I don't think it's a good idea to remove the TODO comment

File: src/platform/common_string_convert.cpp (hunk 2 above)
Comment (devhawk) [reply src/platform/common_string_convert.cpp]: ditto :)

**Comment by devhawk** [comment]
Note, I'm merging this PR w/o @DefaultRyan approval, even though he's the PAL code owner. He's unavailable right now + he actually did most of the PAL changes in a different branch that I cherry picked into this one.

**Pull request merged**